app/api/api_v1/endpoints/search.py

```diff
@@ -1,39 +1,81 @@
 from typing import Any, List
 from fastapi import APIRouter, HTTPException
 from app import schemas
+from app.api import utils
+from app.api import index
 from app.api import deps
 
-from app.shared_state import existing_collections
+from app.core.config import settings  # pylint: disable=C0415
 
 router = APIRouter()
 
+import lxml.html
+import numpy as np
+
+
 @router.get("/", response_model = schemas.SearchResultFull)
```

Comment:
```python
@router.get("/{collection_name} ...."
```

I missed that previously.
We should search in the exact collection. Each deployment of Joan will have a unique collection. There is no reason to return documents from other collections, as for now at least.